### ISSUE TITLE
Complete deterministic training by adding CUBLAS workspace config and DataLoader worker seeding

### ISSUE THREAD

**Issue by abhaygoudannavar** [mllam/neural-lam#265]
The Trainer already sets deterministic=True ([train_model.py#L319](https://github.com/mllam/neural-lam/blob/main/neural_lam/train_model.py#L319)), but two things are still missing for runs to be truly reproducible with the same seed.

1. CUBLAS_WORKSPACE_CONFIG is not set Without this env var, cuBLAS matrix multiplications on GPU can still be non-deterministic even with deterministic=True. PyTorch [documents this explicitly](https://pytorch.org/docs/stable/notes/randomness.html#cuda-convolution-benchmarking). Fix is one line at the top of main():

``` python
os.environ.setdefault("CUBLAS_WORKSPACE_CONFIG", ":4096:8")
``` 
2. No worker_init_fn on the DataLoaders With num_workers=16, each worker gets a copy of the random state. Without a worker_init_fn, workers across runs can end up with correlated seeds. PyTorch [recommends](https://pytorch.org/docs/stable/data.html#randomness-in-multi-process-data-loading) seeding each worker independently:

```python
def _worker_init_fn(worker_id):
    worker_seed = torch.initial_seed() % 2**32
    np.random.seed(worker_seed)
    random.seed(worker_seed)
```
Then pass worker_init_fn=_worker_init_fn to all three DataLoaders in WeatherDataModule.

**Comment by Shalinis19137** [comment]
I’ve reviewed this issue and would like to contribute to it. Kindly assign it to me so I can start working on a fix

**Comment by abhaygoudannavar** [comment]
@Shalinis19137 thanks for showing interest, i have linked a PR: #267  to the issue go through the PR once and suggest any changes needed.

**Comment by joeloskarsson** [comment]
Is the worker seeding not handled by lightning already?
